src/main/java/com/metamx/common/CompressionUtils.java

```diff
@@ -0,0 +1,348 @@
+          }
+
+        }
+    );
+  }
+
+  public static long gunzip(InputStream in, OutputStream out) throws IOException
+  {
+    try (GZIPInputStream gzipInputStream = gzipInputStream(in)) {
+      return ByteStreams.copy(gzipInputStream, out);
+    }
+    finally {
+      out.close();
+    }
+  }
+
+  /**
+   * A gunzip function to store locally
+   *
+   * @param in          The factory to produce input streams
+   * @param outFile     The file to store the result into
+   * @param shouldRetry A predicate to indicate if the Throwable is recoverable
+   *
+   * @return The count of bytes written to outFile
+   */
+  public static FileUtils.FileCopyResult gunzip(
+      final ByteSource in,
+      final File outFile,
+      Predicate<Throwable> shouldRetry
+  )
+  {
+    return FileUtils.retryCopy(
+        new ByteSource()
+        {
+          @Override
+          public InputStream openStream() throws IOException
+          {
+            return gzipInputStream(in.openStream());
+          }
+        },
+        shouldRetry,
+        outFile,
+        10
+    );
+  }
+
+
+  public static FileUtils.FileCopyResult gunzip(final ByteSource in, File outFile)
+  {
+    return gunzip(in, outFile, FileUtils.MD5ERROR);
+  }
+
+  public static long gzip(InputStream inputStream, OutputStream out) throws IOException
+  {
+    try (GZIPOutputStream outputStream = new GZIPOutputStream(out)) {
+      return ByteStreams.copy(inputStream, outputStream);
+    }
+    finally {
+      out.close();
+      inputStream.close();
+    }
+  }
+
+  public static FileUtils.FileCopyResult gzip(final File inFile, final File outDir, Predicate<Throwable> shouldRetry)
+      throws IOException
+  {
+    final File outFile = Paths.get(outDir.getAbsolutePath(), inFile.getName() + GZ_SUFFIX).toFile();
+    FileUtils.retryCopy(
+        new ByteSource()
+        {
+          @Override
+          public InputStream openStream() throws IOException
+          {
+            return new FileInputStream(inFile);
+          }
+        },
+        shouldRetry,
+        new ByteSink()
+        {
+          @Override
+          public OutputStream openStream() throws IOException
+          {
+            return new GZIPOutputStream(new FileOutputStream(outFile));
+          }
+        },
+        10
+    );
+    return FileUtils.FileCopyResult.buildResult(outFile.toPath(), "MD5");
+  }
+
+
+  public static FileUtils.FileCopyResult gzip(final File inFile, final File outDir) throws IOException
+  {
+    return gzip(inFile, outDir, FileUtils.MD5ERROR);
+  }
+
+  public static Boolean isZip(String fName)
+  {
+    if (Strings.isNullOrEmpty(fName)) {
+      return false;
+    }
+    return fName.endsWith(ZIP_SUFFIX); // Technically a file named `.zip` would be fine
+  }
+
+  public static Boolean isGz(String fName)
+  {
+    if (Strings.isNullOrEmpty(fName)) {
+      return false;
+    }
+    return fName.endsWith(GZ_SUFFIX) && fName.length() > GZ_SUFFIX.length();
```

Comment:
It seems a bit strange to try to maintain the same semantics for gzip and zip, since gzip is just a compression format and zip is a container format. tar.gz would be the closest to a zip file.

Are we actually using the gzip stuff anywhere yet?

> On Mar 18, 2015, at 11:50, Charles Allen <notifications@github.com> wrote:
> 
> In src/main/java/com/metamx/common/CompressionUtils.java:
> 
> > +  }
> > +
> > +  public static Boolean isZip(String fName)
> > +  {
> > +    if (Strings.isNullOrEmpty(fName)) {
> > +      return false;
> > +    }
> > +    return fName.endsWith(ZIP_SUFFIX); // Technically a file named `.zip` would be fine
> > +  }
> > +
> > +  public static Boolean isGz(String fName)
> > +  {
> > +    if (Strings.isNullOrEmpty(fName)) {
> > +      return false;
> > +    }
> > +    return fName.endsWith(GZ_SUFFIX) && fName.length() > GZ_SUFFIX.length();
> That is true, but CompressionUtils.gunzip(File, File) takes an output directory to be consistent with CompressionUtils.unzip(File, File)
> 
> Any of the other gunzip methods (stream based) take a target file as the second parameter.
> 
> —
> Reply to this email directly or view it on GitHub.
>